### PR TITLE
Alerting: Adds support for overriding 'dedup_key' via alert tags when using the Pagerduty notifier

### LINKED ISSUE #26863: Alerting: PagerDuty de-duplication key too simple, may cause incorrect incident de-duplication
**What happened**:
I have two Grafana instances in different clusters, both Grafanas have the same PG account and integration key configured. Both instances have the same dashboards configured. (through import-export).

If alerts with the same alert ID but from different Grafana instances are triggered, only one incident shows up in Pager Duty. It is caused by de-duplication on the PG side (note that the same ID can have different alerts).

**What you expected to happen**:
All alerts should be shown in PG. The dedup key should be unique or more complex.

The current dedup key is created by the following scheme: `bodyJSON.Set("dedup_key", "alertId-"+strconv.FormatInt(evalContext.Rule.ID, 10))`.

https://github.com/grafana/grafana/blob/9c81c7aa4462057abec2cc2426d575843cb6b2b4/pkg/services/alerting/notifiers/pagerduty.go#L223

We can probably create a more complex key, something like this: `alerId+dashboardId+panelId`.

**How to reproduce it (as minimally and precisely as possible)**:
1. Create fresh identical Grafana instances in two similar environments.
2. Configure identical PagerDuty notification channels in the Grafana instances. 
3. Create or import dashboards with alerts.
4. Trigger alerts with the same alert ID on the Grafana instances. Check the alert ID through the Alerting API.
5. Check Pager Duty incidents, only one should present. 

**Environment**:

- Grafana version: grafana:6.5.1
- Data source type & version: Prometheus 2.17.2
- OS Grafana is installed on: docker container,
- User OS & Browser: Ubuntu 20.4, firefox
- Grafana plugins: no
- Others: -

### PR DESCRIPTION
**Pull request by alavrovinfb**
- [X] Added ability to override PagerDuty 'dedup_key' via Alert tags

- [X] Updated unit tests

- [X] Updated Documentation

**What this PR does / why we need it**:

Changes from this PR allows user to override weak Grafana generated `dedup_key` on their own.
It can avoid incorrect PagerDuty incidents de-duplication. A possible issue is described [here](https://github.com/grafana/grafana/issues/26863l)

**Which issue(s) this PR fixes**:

This PR has been raised to fix Grafana issue [#26863 ](https://github.com/grafana/grafana/issues/26863)
 

Fixes https://github.com/grafana/grafana/issues/26863

### CODE REVIEW

File: docs/sources/alerting/notifications.md [hunk 1]
```diff
@@ -125,13 +125,14 @@ Severity | Level for dynamic notifications, default is `critical` (1)
 Auto resolve incidents | Resolve incidents in PagerDuty once the alert goes back to ok
 Message in details | Removes the Alert message from the PD summary field and puts it into custom details instead (2)
 
->**Note:** The tags `Severity`, `Class`, `Group`, and `Component` have special meaning in the [Pagerduty Common Event Format - PD-CEF](https://support.pagerduty.com/docs/pd-cef). If an alert panel defines these tag keys, then they are transposed to the root of the event sent to Pagerduty. This means they will be available within the Pagerduty UI and Filtering tools. A Severity tag set on an alert overrides the global Severity set on the notification channel if it's a valid level.
+>**Note:** The tags `Severity`, `Class`, `Group`, `dedup_key` and `Component` have special meaning in the [Pagerduty Common Event Format - PD-CEF](https://support.pagerduty.com/docs/pd-cef). If an alert panel defines these tag keys, then they are transposed to the root of the event sent to Pagerduty. This means they will be available within the Pagerduty UI and Filtering tools. A Severity tag set on an alert overrides the global Severity set on the notification channel if it's a valid level.
```
Comment (achatterjee-grafana) [inline docs/sources/alerting/notifications.md:128]: Doc change looks OK.

Comment (oddlittlebird) [inline docs/sources/alerting/notifications.md:128]:
```suggestion
>**Note:** The tags `Severity`, `Class`, `Group`, `dedup_key`, and `Component` have special meaning in the [Pagerduty Common Event Format - PD-CEF](https://support.pagerduty.com/docs/pd-cef). If an alert panel defines these tag keys, then they are transposed to the root of the event sent to Pagerduty. This means they will be available within the Pagerduty UI and Filtering tools. A Severity tag set on an alert overrides the global Severity set on the notification channel if it's a valid level.
```

File: docs/sources/alerting/notifications.md [hunk 2]
```diff
@@ -125,13 +125,14 @@ Severity | Level for dynamic notifications, default is `critical` (1)
...
 
 >Using Message In Details will change the structure of the `custom_details` field in the PagerDuty Event.
 This might break custom event rules in your PagerDuty rules if you rely on the fields in `payload.custom_details`.
 Move any existing rules using `custom_details.myMetric` to `custom_details.queries.myMetric`.
 This behavior will become the default in a future version of Grafana.
 
+>Using `dedup_key` tag will override grafana generated `dedup_key` with a custom key.
```
Comment (oddlittlebird) [inline docs/sources/alerting/notifications.md:135]:
```suggestion
> Using `dedup_key` tag will override grafana generated `dedup_key` with a custom key.
```

**Comment by oddlittlebird** [comment]
I made some suggestions on the docs.

**Review by marefr — APPROVED** [review]
LGTM

**Comment by marefr** [comment]
Thank you for contributing to Grafana!

**Pull request merged**